vision/fiducial_detection_and_pose.cc

```diff
@@ -0,0 +1,81 @@
+#include "vision/fiducial_detection_and_pose.hh"
+#include <cassert>
+#include <cstdlib>
+
+
+namespace jet {
+
+namespace {
+
+MarkerRvecsTvecs rvecs_tvecs_from_corners(const std::vector<std::vector<cv::Point2f>>& corners) {
+  std::vector<cv::Vec3d> rvecs, tvecs;
+  const cv::Mat camera_matrix = (cv::Mat1d(3, 3) << 320, 0, 320, 0, 320, 320, 0, 0, 1);
+  const cv::Mat distortion_coefficients =
+      (cv::Mat1d(1, 8) << 0, 0, 0, 0, 0, 0, 0, 0);
+  cv::aruco::estimatePoseSingleMarkers(corners, 0.49375, camera_matrix,
+                                       distortion_coefficients, rvecs, tvecs);
+  MarkerRvecsTvecs result;
+  result.rvecs = rvecs;
+  result.tvecs = tvecs;
+  return result;
+}
+
+}
+
+std::vector<MarkerDetection> detect_markers(const cv::Mat& inputImage) {
+  cv::Ptr<cv::aruco::Dictionary> dictionary =
+      cv::aruco::getPredefinedDictionary(cv::aruco::DICT_6X6_250);
+  std::vector<int> ids;
+  std::vector<std::vector<cv::Point2f>> corners;
+  cv::aruco::detectMarkers(inputImage, dictionary, corners, ids);
+  MarkerRvecsTvecs rvecs_tvecs = rvecs_tvecs_from_corners(corners);
+
+  for (auto const& quad : corners) {
+    for (auto const& center : quad) {
+      cv::circle(inputImage, center, 10, cv::Scalar(255, 0, 0));
+    }
+  }
```

Comment:
Do one of:
- Include a debug flag as an argument for disabling this drawing
- Put this in a different function and call that function when you want to debug

This code also almost surely doesn't compile!